src/main/java/com/sap/cloudfoundry/client/facade/rest/CloudControllerRestClientImpl.java

```diff
@@ -1307,36 +1297,46 @@ public void updateApplicationMetadata(UUID guid, org.cloudfoundry.client.v3.Meta
     @Override
     public void updateApplicationStaging(String applicationName, Staging staging) {
         UUID applicationGuid = getRequiredApplicationGuid(applicationName);
-        UpdateApplicationRequest updateApplicationRequest = UpdateApplicationRequest.builder()
-                                                                                    .applicationId(applicationGuid.toString())
-                                                                                    .lifecycle(buildApplicationLifecycle(staging))
-                                                                                    .build();
         delegate.applicationsV3()
-                .update(updateApplicationRequest)
+                .update(UpdateApplicationRequest.builder()
+                                                .applicationId(applicationGuid.toString())
+                                                .lifecycle(buildApplicationLifecycle(staging))
+                                                .build())
                 .block();
         GetApplicationProcessResponse applicationProcess = getApplicationProcess(applicationGuid);
         updateApplicationProcess(applicationGuid, staging, applicationProcess);
     }
 
     @Override
     public void updateApplicationRoutes(String applicationName, Set<CloudRouteSummary> updatedRoutes) {
-        CloudApplication application = getApplication(applicationName);
-        Set<CloudRouteSummary> currentRoutes = application.getRoutes();
+        UUID applicationGuid = getApplicationGuid(applicationName);
+        List<RouteResource> appRoutes = getRouteResourcesByAppGuid(applicationGuid).collectList()
+                                                                                   .block();
+
+        List<RouteResource> outdatedRoutes = filterRoutes(appRoutes, updatedRoutes);
 
-        Set<CloudRouteSummary> outdatedRoutes = selectRoutesOnlyInFirst(currentRoutes, updatedRoutes);
+        Set<CloudRouteSummary> newRoutes = filterRoutes(updatedRoutes, appRoutes);
 
-        Set<CloudRouteSummary> newRoutes = selectRoutesOnlyInFirst(updatedRoutes, currentRoutes);
+        removeRoutes(outdatedRoutes, applicationGuid);
+        addRoutes(newRoutes, applicationGuid);
+    }
 
-        removeRoutes(outdatedRoutes, application.getMetadata()
-                                                .getGuid());
-        addRoutes(newRoutes, application.getMetadata()
-                                        .getGuid());
+    private Set<CloudRouteSummary> filterRoutes(Set<CloudRouteSummary> allRoutes, List<RouteResource> routeResources) {
```

Comment:
I am little bit lost between two `filterRoutes` methods. Both do some filter operations and return different results. Think about more explicit name of methods. Is it possible to merge them into one?

Reply:
Renamed to `getOutdatedRoutes` and `getNewRoutes`